lib/sober_swag/reporting/input/in_range.rb

```diff
@@ -0,0 +1,61 @@
+module SoberSwag
+  module Reporting
+    module Input
+      ##
+      # Specify that an item must be within a given range in ruby.
+      # This gets translated to `minimum` and `maximum` keys in swagger.
+      #
+      # This works with endless ranges (Ruby 2.6+) and beginless ranges (Ruby 2.7+)
+      class InRange < Base
+        def initialize(input, range)
+          @input = input
+          @range = range
+        end
+
+        ##
+        # @return [Interface]
+        attr_reader :input
+
+        ##
+        # @return [Range]
+        attr_reader :range
+
+        ##
+        # @return [Range]
+        def call(value)
+          res = input.call(value)
+
+          return res if res.is_a?(Report::Base)
+          return Report::Value.new(['was not in minimum/maximum range']) unless range.member?(res)
+
+          res
+        end
+
+        def swagger_schema
+          schema, found = input.swagger_schema
+
+          merged =
+            if schema.key?(:$ref)
+              { allOf: [schema] }
+            else
+              schema
+            end.merge(maximum_portion).merge(minimum_portion)
+
+          [merged, found]
+        end
+
+        def maximum_portion
+          return {} unless range.end
```

Comment:
Not a blocker but just want to check
Are we expecting boolean here if so can we do `range.end?`

Reply:
This is to check if it's nil or not! There's no `#end?` method defined on ranges.